vic/drivers/shared_all/src/generate_default_state.c

```diff
@@ -125,6 +133,34 @@ generate_default_state(all_vars_struct *all_vars,
         }
     }
 
+
+    /************************************************************************
+       Initialize gridcell-averaged albedo
+    ************************************************************************/
+    // vegetation class-weighted albedo over gridcell
+    albedo_sum = 0;
+    for (veg = 0; veg <= Nveg; veg++) {
+        Cv = veg_con[veg].Cv;
+        if (Cv > 0) {
+            for (band = 0; band < options.SNOW_BAND; band++) {
+                if (soil_con->AreaFract[band] > 0.) {
+                    // TO-DO: account for treeline and lake factors
+                    AreaFactor = (Cv * soil_con->AreaFract[band] *
+                                  TreeAdjustFactor * lakefactor);
+                    // cold start, so assuming bare (free of snow) albedo
+                    if (veg != Nveg) {
+                        albedo_sum += AreaFactor * veg_var[veg][band].albedo;
+                    }
+                    else {
+                        // this is the bare soil class, so use bare soil albedo parameter
+                        albedo_sum += AreaFactor * param.ALBEDO_BARE_SOIL;
+                    }
+                }
+            }
+        }
+    }
+    all_vars->gridcell_avg.avg_albedo = albedo_sum;
```

Comment:
See previous comment: Either use the local variable `gridcell_avg` (but change to a pointer) and use

``` C
gridcell_avg->avg_albedo = albedo_sum;
```
or get rid of local variable and just leave this as is.